File: UseCases/src/test/resources/dummy/ScopeProtocolTest.swift
open var variable: Object {
set {
}
get {
fatalError("Dummy implementation")
}
}
open subscript(a: Int) -> Int {
set {
}
get {
return 0
}
}
open func method(param: Object, closure: () -> ()) -> Object {
fatalError("Dummy implementation")
}

File: src/test/kotlin/codes/seanhenry/mockgenerator/generator/templates/ScopeProtocolTest.kt
package codes.seanhenry.mockgenerator.generator.templates

import codes.seanhenry.mockgenerator.entities.Method
import codes.seanhenry.mockgenerator.entities.Property
import codes.seanhenry.mockgenerator.generator.MockTransformer

class ScopeProtocolTest : MockGeneratorTestTemplate() {

  override fun build(generator: MockTransformer) {
    generator.setScope("open")
    generator.add(
        Property.Builder("variable")
            .type("Object")
            .build()
    )
    generator.add(
        Method.Builder("method")
            .parameter("param") { it.type("Object") }
            .parameter("closure") { it.type().function { } }
            .returnType("Object")
            .build()
    )
  }
}
